### ISSUE TITLE
Now，Is there an operational interface for setting extended xattrs between NFS clients and server?

### ISSUE THREAD

**Issue by comosus** [nfs-ganesha/nfs-ganesha#526]
hi,
  In ganesha, is/are there interface(s) for setting xattrs in ganesha.client? And it goes well in ganesha.server. 
thanks.

**Comment by dang** [comment]
Ganesha supports RFC8276 for XATTR over NFSv4, but only FSAL_GPFS has implemented it, so other FSALs will just return ENOTSUP to the client.  Someone would have to implement this for other FSALs.

**Comment by phdeniel** [comment]
Hi Dan,

it would be great o have NFSv4 XATTR implemented for FSAL_VFS, for 
xattrs are a posix feature (I have coded support for them in the 9p code 
for NFS-Ganesha). Since SELinux is "xattr based" too (labels for files 
are stored in special xattr security.selinux)  it would open an 
interesting path to SELinux support.

     Regards

         Philippe

**Comment by dang** [comment]
I agree it'd be nice, but it's not currently on my radar to implement.

**Comment by comosus** [comment]
> Ganesha supports RFC8276 for XATTR over NFSv4, but only FSAL_GPFS has implemented it, so other FSALs will just return ENOTSUP to the client. Someone would have to implement this for other FSALs.

Ok, I get it now, thanks

**Comment by ffilz** [comment]
Let's keep this one for backlog.

Note that xattr for FSAL_GLUSTER is in progress.

**Comment by ffilz** [comment]
xattr support has been added to CEPH, it still could be done for VFS if someone wanted to pick that up

**Comment by ffilz** [comment]
xattr support is available for the following FSALs: CEPH, GLUSTER, GPFS, and RGW.

Notably it is not available for PROXY_V3, PROXY_V4, or VFS.

I'm not sure how important it is for FSALs other than the ones implemented in, Adding xattr support to FSAL_VFS might have value as a reference implementation for ease of testing.

**Comment by ffilz** [comment]
FSAL_VFS has had xattr support for some time, so now it's just down to the currently not actively supported PROXY FSALs and some of the other contributed FSALs.
